mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -241,8 +252,77 @@ pub async fn process_msg(ctx: MmArc, topic: &str, msg: &[u8]) {
     }
 }
 
+pub async fn process_watcher_msg(ctx: MmArc, msg: &[u8]) {
+    let msg = match decode_signed::<SwapWatcherMsg>(msg) {
+        Ok(m) => m,
+        Err(watcher_msg_err) => {
+            error!("Couldn't deserialize 'SwapWatcherMsg': {:?}", watcher_msg_err);
+            // Drop it to avoid dead_code warning
+            drop(watcher_msg_err);
+            return;
+        },
+    };
+
+    match msg.0 {
+        SwapWatcherMsg::TakerSwapWatcherMsg(watcher_data) => spawn_taker_swap_watcher(ctx, *watcher_data),
+    }
+}
+
+fn spawn_taker_swap_watcher(ctx: MmArc, watcher_data: TakerSwapWatcherData) {
+    let swap_ctx = SwapsContext::from_ctx(&ctx).unwrap();
+    let msgs = swap_ctx.swap_msgs.lock().unwrap();
+    let mut taker_swap_watchers = swap_ctx.taker_swap_watchers.lock().unwrap();
+    // Return if taker, maker or watcher swap already exists
+    // This needs discussion
+    if msgs.contains_key(&watcher_data.uuid) || taker_swap_watchers.contains(&watcher_data.uuid) {
+        return;
+    }
+    taker_swap_watchers.insert(watcher_data.uuid);
```

Comment:
It's too early to mark the watcher loop as running here. It might not start if coins are not activated, but they can be enabled in between of receiving `SwapWatcherMsg` rebroadcasts. In this case, node is able to `run_watcher`, but it won't.